Gitweb:     http://git.kernel.org/linus/cd5a2685de4a642fd0bd763e8c19711ef08dbe27
Commit:     cd5a2685de4a642fd0bd763e8c19711ef08dbe27
Parent:     6be7d3062b59af891be7e40c6802350de5f78cef
Author:     Marcelo Tosatti <mtosatti@redhat.com>
AuthorDate: Sat Oct 17 22:47:23 2009 -0300
Committer:  Avi Kivity <avi@redhat.com>
CommitDate: Thu Dec 3 09:32:19 2009 +0200

    KVM: fix irq_source_id size verification
    
    find_first_zero_bit works with bit numbers, not bytes.
    
    Fixes
    
    https://sourceforge.net/tracker/?func=detail&aid=2847560&group_id=180599&atid=893831
    
    Reported-by: "Xu, Jiajun" <jiajun.xu@intel.com>
    Cc: stable@kernel.org
    Signed-off-by: Marcelo Tosatti <mtosatti@redhat.com>
---
 virt/kvm/irq_comm.c |    7 +++----
 1 files changed, 3 insertions(+), 4 deletions(-)

--- a/virt/kvm/irq_comm.c
+++ b/virt/kvm/irq_comm.c
@@ -205,10 +205,9 @@ int kvm_request_irq_source_id(struct kvm
 	int irq_source_id;
 
 	mutex_lock(&kvm->irq_lock);
-	irq_source_id = find_first_zero_bit(bitmap,
-				sizeof(kvm->arch.irq_sources_bitmap));
+	irq_source_id = find_first_zero_bit(bitmap, BITS_PER_LONG);
 
-	if (irq_source_id >= sizeof(kvm->arch.irq_sources_bitmap)) {
+	if (irq_source_id >= BITS_PER_LONG) {
 		printk(KERN_WARNING "kvm: exhaust allocatable IRQ sources!\n");
 		return -EFAULT;
 	}
@@ -228,7 +227,7 @@ void kvm_free_irq_source_id(struct kvm *
 
 	mutex_lock(&kvm->irq_lock);
 	if (irq_source_id < 0 ||
-	    irq_source_id >= sizeof(kvm->arch.irq_sources_bitmap)) {
+	    irq_source_id >= BITS_PER_LONG) {
 		printk(KERN_ERR "kvm: IRQ source ID out of range!\n");
 		return;
 	}
